python/ray/_private/usage/usage_lib.py

```diff
@@ -158,6 +161,62 @@ class UsageStatsEnabledness(Enum):
 _recorded_library_usages = set()
 
 
+# NOTE: Do not change the write / read protocol. That will cause
+# version incompatibility issues.
+class LibUsageRecorder:
+    """A class to put/get the library usage to the ray tmp folder.
+
+    See https://github.com/ray-project/ray/pull/25842 for more details.
+    """
+
+    def __init__(self, temp_dir_path: str):
+        self._lib_usage_dir = Path(temp_dir_path)
+        self._lib_usage_prefix = "_ray_lib_usage-"
+        self._lib_usage_filename_match = re.compile(
+            f"{self._lib_usage_prefix}([0-9a-zA-Z_]+).txt"
+        )
+
+    def put_lib_usage(self, lib_name: str):
+        """Put the library usage to the ray tmp folder."""
+        if sys.platform == "win32":
+            # In Linux, exclusive file open is supported.
+            # However, it is not clear if it is supported on Windows,
+            # so we just use a file lock.
+            with TempFileLock(str(self.lib_usage_file), timeout=2):
```

Comment:
Remove temp file lock.

Reply:
This is just for Windows. It seems like exclusive file creation is not supported by Windows, so it might be safer to just use file lock there?